### PR TITLE
implement expand

### PR DESCRIPTION
**Pull request by Mei0211**
@BruceDai @huningxin @fdwr @fujunwei PTAL, thanks.

### CODE REVIEW

File: test/expand_test.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+describe('test expand', function() {
+  function testExpand(input, newShape, expected) {
+    const tensor = new Tensor(input.shape, input.data);
+    const outputTensor = expand(tensor, newShape);
+    utils.checkShape(outputTensor, expected.shape);
+    utils.checkValue(outputTensor, expected.data);
+  }
+
+  it('expand changed dimensions', function() {
```
Comment (BruceDai) [inline test/expand_test.js:15]: Please add a more test for testing 0D expand to ND, you can refer to https://source.chromium.org/chromium/chromium/src/+/main:services/webnn/dml/graph_impl_test.cc;l=1456, thanks.

**Review by BruceDai — COMMENTED** [review]
Thanks @Mei0211! 
Let's add a more test for testing 0D expand to ND.

**Review by huningxin — COMMENTED** [review]
lgtm % other reviewers' comments, thanks!

File: test/expand_test.js (hunk 1 above)
Comment (fdwr) [reply test/expand_test.js:15]: I included some cases [here](https://github.com/webmachinelearning/webnn/pull/478#discussion_r1396805359), where new shape is longer and shorter than input.rank (which is perfectly fine, so long as they are broadcast compatible with each other):

input.shape = [2,1,4], newShape = [3,1], output.shape = [2,3,4]
input.shape = [2,1,4], newShape = [5,1,3,4], output.shape = [5,2,3,4]
input.shape = [], newShape = [2,3], output.shape = [2,3]
input.shape = [2,3], newShape = [], output.shape = [2,3]

**Review by fdwr — APPROVED** [review]
Barring remaining comments, LGTM. TY.

**Comment by Mei0211** [comment]
I have modified the expand function and added more cases. PTAL,thanks!

File: src/expand.js [hunk 2]
```diff
@@ -10,5 +11,13 @@ import {broadcast} from './lib/broadcast.js';
...
+  if (input.shape.length === 0) {
+    const inputReshape = new Tensor([1], input.data);
+    const outputShape = getBroadcastShape(inputReshape.shape, newShape);
+    return broadcast(inputReshape, outputShape);
+  } else {
+    const inputReshape = new Tensor(input.shape, input.data);
+    const outputShape = getBroadcastShape(inputReshape.shape, newShape);
+    return broadcast(inputReshape, outputShape);
+  }
 }
```
Comment (BruceDai) [inline src/expand.js:23]: You may optimize this function as below
```js
export function expand(input, newShape) {
  const inputReshape = input.shape.length === 0 ? new Scalar(input.data) : input;
  const outputShape = getBroadcastShape(inputReshape.shape, newShape);
  return broadcast(inputReshape, outputShape);
}
```

File: test/expand_test.js (hunk 1 above)
Comment (Mei0211) [reply test/expand_test.js]: I have added these cases. PTAL:)

File: src/expand.js (hunk 2 above)
Comment (Mei0211) [reply src/expand.js:19]: Thanks! I have optimized this function!

**Comment by BruceDai** [comment]
LGTM, thanks @Mei0211

**Pull request merged**